arch/rv32i/src/plic.rs

```diff
@@ -63,6 +66,14 @@ pub unsafe fn disable_all() {
     }
 }
 
+/// This is a generic implementation. There may be board specific versions as
+/// some platforms have added more bits to the `mtvec` register.
+pub unsafe fn surpress_all() {
```

Comment:
Ditto, I don't think this needs to be `unsafe`. If the purpose is prevent general code from using it, a better strategy would be to use visibility---either changing the function visibility, or wrapping making these functions methods of a type and only giving access to that type to the appropriate clients (e.g. by making the _constructor_ `unsafe`).